src/components/report-viewer/report-viewer.jsx

```diff
@@ -56,59 +62,57 @@ export default function ReportViewer({ report, reportType }) {
     }, []);
 
     const refreshLogsOnSelectedReport = useCallback(
-        (reportId, severityFilter) => {
-            let severityList = [];
-            for (const [severity, selected] of Object.entries(severityFilter)) {
-                if (selected) {
-                    severityList.push(severity);
+        (selectedReportId, severityFilter, messageFilter) => {
+            //we need to do this check because selectedReportId can be outdated when switching between the different place we have this component
+            if (reportTreeData.current[selectedReportId] != null) {
+                let severityList = [];
+                for (const [severity, selected] of Object.entries(severityFilter)) {
+                    if (selected) {
+                        severityList.push(severity);
+                    }
                 }
-            }
-
-            if (severityList.length === 0) {
-                // no severity => there is no log to fetch, no need to request the back-end
-                setSelectedReportId(reportId);
-                setLogs([]);
-                setHighlightedReportId(null);
-                return;
-            }
-
-            fetchLogs(reportId, severityList, reportTreeData.current[reportId].type).then((logs) => {
-                if (logs !== undefined) {
-                    setLogs(logs);
-                    setSelectedReportId(reportId);
+                if (severityList.length === 0) {
+                    setLogs([]);
                     setHighlightedReportId(null);
+                    return;
                 }
-            });
+                fetchReportLogs(
+                    selectedReportId,
+                    severityList,
+                    reportTreeData.current[selectedReportId].type,
+                    messageFilter
+                ).then((reportLogs) => {
+                    setLogs(mapReportLogs(reportLogs));
+                });
+            }
         },
-        [fetchLogs]
+        [fetchReportLogs]
     );
 
+    useEffect(() => {
+        refreshLogsOnSelectedReport(selectedReportId, severityFilter, messageFilter);
+    }, [messageFilter, severityFilter, selectedReportId, refreshLogsOnSelectedReport]);
+
     useEffect(() => {
         const reportTree = mapReportsTree(report);
         treeView.current = initializeTreeDataAndComponent(reportTree);
-        setSelectedReportId(report.id);
         setExpandedTreeReports([report.id]);
         setLogs(mapReportLog(report, reportTree.severities));
```

Comment:
You can remove that then the method mapReportLog will be unused so you can delete it